cmd/ci-secret-generator/README.md

```diff
@@ -31,6 +31,22 @@ create two bitwarden entries - 'first_item' and 'second_item'
 
 * `field2` of `second_item`, would be `field2_contents` in Bitwarden with item-name `second_item`
 
+Parameters can be passed in to decrease repetition in the config file by adding the params dictionary in the config file.
+Eg.
+```yaml
+- item_name: item$(cluster)$(env)
+  fields:
+    - name: field1
+      cmd: echo "$(cluster) $(env)"
+  params:
+    cluster:
+      - build01
+      - build02
+    env:
+      - prod
+      - stagin
+```
+This would create four items with itemnames - itembuild01prod, itembuild02prod,itembuild01env, itembuild02env and the corresponding field1 which would contain the output of the corresponding echo, where the $(paramname) would be replaced with the values of the corresponding paramname.
```

Comment:
This approach is a little fraught - why do we assume that the combinatorics should be a full cross product between the input parameter lists? IMHO a more resilient approach is:

```yaml
params:
- cluster: build01
  env: prod
- cluster: build02
  env: prod
...
```

I'm also not super sure that this is the most straightforward, transparent approach - doesn't YAML have templating/anchors built in that can do this for us?

Reply:
We can do it in that manner as well, but taking the cross-product is what I and @alvaroaleman  had discussed and decided on to reduce the stuff which would need to be explicitly typed out by the user(which was the overall intention of this change). 
Regarding YAML templating I am not aware of anything and my initial search only gave me some third party tools and default go templating which I'm not sure would serve our purpose directly. Is there something specific you had in mind?